src/SymmetricPositiveDefinite.jl

```diff
@@ -0,0 +1,174 @@
+function distance(P::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric},x,y)
+    s = real.( eigen( x,y ).values )
+    return any(s .<= eps() ) ? 0 : sqrt(  sum( abs.(log.(s)).^2 )  )
+end
+
+inner(P::SymmetricPositiveDefinite{N}, x, w, v) = inner(MetricManifold(P,LinearAffineMetric),x,w,v)
+function inner(P::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric},x,w,v)
+	svd1 = svd(x)
+	U = svd1.U
+	S = svd1.S
+	SInv = Diagonal( 1 ./ S )
+	return tr( w * U * SInv * transpose(U) * v * U * SInv * transpose(U) )
+end
+
+norm(P::SymmetricPositiveDefinite{N},x,v) = sqrt( inner(P,x,v,v) )
+norm(P::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric},x,v) = sqrt( inner(P,x,v,v) )
+
+function exp!(P::SymmetricPositiveDefinite{N},y,x,v) = exp!(MetricManifold(SymmetricPositiveDefinite,LinearAffineMetric),y,x,v)
+function exp!(P::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric}, y, x, v)
+	svd1 = svd(x)
+	U = svd1.U
+	S = svd1.S
+    Ssqrt = Diagonal( sqrt.(S) )
+    SsqrtInv = Diagonal( 1 ./ sqrt.(S) )
+    xSqrt = U*Ssqrt*transpose(U);
+    xSqrtInv = U*SsqrtInv*transpose(U)
+    T = xSqrtInv * (t.*ξ.value) * xSqrtInv
+    eig1 = eigen(0.5*( T + transpose(T) ) ) # numerical stabilization
+   	Se = Diagonal( exp.(eig1.values) )
+    Ue = eig1.vectors
+    y = xSqrt*Ue*Se*transpose(Ue)*xSqrt
+    y = 0.5*( y + transpose(y) ) # numerical stabilization
+    return y
+end
+
+function log!(P::SymmetricPositiveDefinite{N}, v, x, y) = exp!(MetricManifold(P,LinearAffineMetric),y,x,v)
+function log!(P::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric}, v, x, y)
+    svd1 = svd( x )
+	U = svd1.U
+	S = svd1.S
+    Ssqrt = Diagonal( sqrt.(S) )
+    SsqrtInv = Diagonal( 1 ./ sqrt.(S) )
+    xSqrt = U*Ssqrt*transpose(U)
+    xSqrtInv = U*SsqrtInv*transpose(U)
+    T = xSqrtInv * getValue(y) * xSqrtInv
+	svd2 = svd(0.5*(T+transpose(T)))
+	Se = Diagonal( log.(max.(svd2.S,eps()) ) )
+	Ue = svd2.U
+	v = xSqrt * Ue*Se*transpose(Ue) * xSqrt
+	v = 0.5*( v + transpose(v) )
+    return v
+end
+
+function vector_transport!(M::SymmetricPositiveDefinite,vto, x, v, y, ::ParallelTransport)
+  if norm(x-y)<1e-13
+    vto = v
+    return vto
+  end
+  svd1 = svd(x)
+  U = svd1.U
+  S = svd1.S
+  Ssqrt = sqrt.(S)
+  SsqrtInv = Diagonal( 1 ./ Ssqrt )
+  Ssqrt = Diagonal( Ssqrt )
+  xSqrt = U*Ssqrt*transpose(U)
+  xSqrtInv = U*SsqrtInv*transpose(U)
+  tv = xSqrtInv * v * xSqrtInv
+  ty = xSqrtInv * y * xSqrtInv
+  svd2 = svd( 0.5*( ty + transpose(ty) ) )
+  Se = Diagonal( log.(svd2.S) )
+  Ue = svd2.U
+  ty2 = Ue*Se*transpose(Ue)
+  eig1 = eigen(  0.5 * (ty2 + transpose(ty2) )  )
+  Sf = Diagonal( exp.(eig1.values) )
+  Uf = eig1.vectors
+  vto = xSqrt*Uf*Sf*transpose(Uf)*(0.5*(tv+transpose(tv)))*Uf*Sf*transpose(Uf)*xSqrt
+  return vto
+end
+
+injectivity_radius(P::SymmetricPositiveDefinite, args...) = Infπ
+
+zero_tangent_vector(P::SymmetricPositiveDefinite, x) = zero(x)
```

Comment:
`zero_tangent_vector!` in `Manifolds.jl` should probably be defined using the `IsDecoratorManifold` trait so that the below lines aren't necessary. I don't know that `zero_tangent_vector` are necessary, since the defaults shouldn't be much less efficient than here (`zero(x)` just calls `fill!(similar(x), zero(eltype(x)))`)